### PR TITLE
chore(ci): Fix the datadog-{agent,traces} integration tests to run in vdev

### PR DESCRIPTION
**Pull request by bruceg**

### CODE REVIEW

File: scripts/integration/datadog-agent/compose.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ services:
     volumes:
     - ../../../tests/data/datadog-agent/conf.yaml:/etc/datadog-agent/conf.d/test.d/conf.yaml
   datadog-trace-agent:
-    image: docker.io/datadog/agent:7.31.0
+    image: docker.io/datadog/agent:${DATADOG_AGENT_VERSION}
```
Comment (bruceg) [inline scripts/integration/datadog-agent/compose.yaml:20]: I'm not sure if this is the right change. Should both images be the same version, or is there a reason we had them separate?

Comment (neuronull) [reply scripts/integration/datadog-agent/compose.yaml:20]: <grumble grumble> I'm fairly sure @jszwedko and I talked about this when I was adding integration tests for the APM stats to the dd traces tests...

I want to say there was a reason we kept it back. But I can't remember.

However, in a general sense, we elected to keep the traces one to the latest offical agent version, in order help catch changes in behavior that we need to adapt vector to. So I think that would ideally be what we'd do for the agent source too. I want to say it had something to do with the api version we were using by default in the agent source, and maybe something with the Agent's default api endpoint for specific 7.x versions. But its fuzzy and I'm not seeing notes about it in the code.

**Review by spencergilbert — APPROVED** [review]
LGTM with @neuronull's approval.

**Comment by neuronull** [comment]
I think the dd integration test is failing ...

File: scripts/integration/docker-compose.datadog-traces.yml [hunk 2]
```diff
@@ -1,91 +0,0 @@
-version: "3"
-
-services:
-  datadog-trace-agent:
```
Comment (spencergilbert) [inline scripts/integration/docker-compose.datadog-traces.yml:4]: @neuronull we still need both this and `datadog-trace-agent-to-vector` for integration tests to pass?

Comment (neuronull) [reply scripts/integration/docker-compose.datadog-traces.yml:4]: I have a feeling reverting the line in question here:

https://github.com/vectordotdev/vector/pull/16130#discussion_r1087048606

, and the change to [scripts/integration/datadog-agent/test.yaml](https://github.com/vectordotdev/vector/pull/16130/files#diff-5df4230c15b5bb57516aff1965f11a5d726bd5031414d4e5459e822c8eb3acdb)

, is what is needed.

I think the deletion of this file you pointed to is just part of moving it over to vdev, if I'm not mistaken?

**Review by neuronull — APPROVED** [review]

**Pull request merged**